corehq/apps/importer/util.py

```diff
@@ -404,12 +407,16 @@ def populate_updated_fields(config, columns, row, datemode):
 
         if update_value is not None:
             if field_map[key]['type_field'] == 'date':
-                update_value = parse_excel_date(update_value, datemode)
-            elif field_map[key]['type_field'] == 'integer':
                 try:
-                    update_value = str(int(update_value))
-                except ValueError:
-                    update_value = ''
+                    update_value = parse_excel_date(update_value, datemode)
+                except Exception:
+                    raise InvalidDateException(key)
+            elif field_map[key]['type_field'] == 'integer':
+                if str(update_value).strip() != '':
```

Comment:
Is this going to handle blanks correctly? Say you're updating a case and want to blank out an integer field, this looks like it will skip updating that case.